Sources/PackageLoading/ModuleMapGenerator.swift

```diff
@@ -71,6 +71,49 @@ public struct ModuleMapGenerator {
         self.diagnostics = diagnostics
     }
 
+    /// Determine the type of module map that applies to the specifed public-headers directory.
+    public static func determineModuleMapType(includeDir: AbsolutePath, moduleName: String, fileSystem: FileSystem) -> ModuleMapType {
```

Comment:
What do you think about refactoring and moving the type computation logic from `generateModuleMap` so we don't have two implementations?

Reply:
Absolutely.  This was always the plan, and was noted in the PR description.  But the intent was to do it in two phases so that one could be more safely pulled back to 5.3, with the switch for SwiftPM to use the module type being for the next release.  But I think it's probably too late to nominate for 5.3 in any case, so I've converted this back to a draft PR until I've put the second part in place.

But to be clear, it was never the intent to leave both implementations in the long run.  :)

Reply:
I've updated the PR to do this.